src/reactor/mod.rs

```diff
@@ -544,6 +548,15 @@ impl Inner {
         debug!("set a timeout: {}", token);
     }
 
+    fn unregister_timeout(&mut self, token: usize) {
+        let pair = &mut self.timeouts[token];
+        if let Some(slot) = pair.0.take() {
+            self.timer_heap.remove(slot);
+        }
+        pair.1 = TimeoutState::NotFired;
```

Comment:
Should this be modified? `update_timeout` doesn't touch it so it seems like we might want to leave it alone.